include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -584,38 +594,36 @@ class NBL_API CBlitImageFilter : public CImageFilter<CBlitImageFilter<Swizzle,Di
+			intermediateExtent[2] = core::vectorSIMDi32(state->outExtent.width, state->outExtent.height, state->outExtent.depth);
 		}
 		// the blit filter will filter one axis at a time, hence necessitating "ping ponging" between two scratch buffers
 		static inline uint32_t getScratchOffset(const state_type* state, bool secondPong)
 		{
 			const auto inType = state->inImage->getCreationParameters().type;
-			const auto kernelX = state->contructScaledKernel(state->kernelX);
-			const auto kernelY = state->contructScaledKernel(state->kernelY);
-			const auto kernelZ = state->contructScaledKernel(state->kernelZ);
+			const auto scaledKernelX = blit_utils_t::constructScaledKernel(state->kernelX, state->inExtentLayerCount, state->outExtentLayerCount);
+			const auto scaledKernelY = blit_utils_t::constructScaledKernel(state->kernelY, state->inExtentLayerCount, state->outExtentLayerCount);
+			const auto scaledKernelZ = blit_utils_t::constructScaledKernel(state->kernelZ, state->inExtentLayerCount, state->outExtentLayerCount);
 
-			const auto window_end = getWindowEnd(state->inImage->getCreationParameters().type,kernelX,kernelY,kernelZ);
+			const auto real_window_size = blit_utils_t::getRealWindowSize(inType,scaledKernelX,scaledKernelY,scaledKernelZ);
 			// TODO: account for the size needed for coverage adjustment
 			// the first pass will be along X, so new temporary image will have the width of the output extent, but the height and depth will need to be padded
 			// but the last pass will be along Z and the new temporary image will have the exact dimensions of `outExtent` which is why there is a `core::max`
-			auto texelCount = state->outExtent.width*core::max<uint32_t>((state->inExtent.height+window_end[1])*(state->inExtent.depth+window_end[2]),state->outExtent.height*state->outExtent.depth);
-			// the second pass will result in an image that has the width and height equal to `outExtent`
+
+			core::vectorSIMDi32 intermediateExtent[3];
+			getIntermediateExtents(intermediateExtent, state, real_window_size);
+
+			assert(intermediateExtent[0].x == intermediateExtent[2].x);
+			auto texelCount = intermediateExtent[0].x * core::max<uint32_t>(intermediateExtent[0].y*intermediateExtent[0].z, intermediateExtent[2].y * intermediateExtent[2].z);
 			if (secondPong)
-				texelCount += core::max<uint32_t>(state->outExtent.width*state->outExtent.height*(state->inExtent.depth+window_end[2]),(state->inExtent.width+window_end[0])*std::thread::hardware_concurrency()*VectorizationBoundSTL);
+				texelCount += core::max<uint32_t>(intermediateExtent[1].x * intermediateExtent[1].y * intermediateExtent[1].z, (state->inExtent.width + real_window_size[0]) * std::thread::hardware_concurrency() * VectorizationBoundSTL);
 			// obviously we have multiple channels and each channel has a certain type for arithmetic
-			return texelCount*MaxChannels*sizeof(value_type);
+			return texelCount*blit_utils_t::MaxChannels*sizeof(value_type);
 		}
```

Comment:
Clean up the sphagetti
https://github.com/Devsh-Graphics-Programming/Nabla/pull/140#discussion_r868150073